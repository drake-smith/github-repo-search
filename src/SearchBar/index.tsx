import React from 'react';
import styled from 'styled-components';
import { media } from '../shared/media';

const { tablet } = media;

interface SearchBarProps {
  searchTerm: string;
  handleSearchTermChange: (searchTerm: string) => void;
  handleSearchSubmit: () => void;
  isDisabled: boolean;
}

const SearchBarForm = styled.form`
  margin-top: 1rem;

  ${tablet`
    margin-top: 2rem;
  `}
`;

const SearchInput = styled.input`
  padding: 0.5rem;
  margin-top: 1rem;
  width: 100%;
  border: 1px solid #202020;

  ${tablet`
    padding: 1rem;
    width: 350px;
  `}

  &:focus, &:hover {
    outline: 2px solid #58bab9;
    outline-offset: -2px;
  }
`;

const SearchButton = styled.input<{ disabled: boolean }>`
  margin-top: 0.5rem;
  color: #fff;
  width: 150px;
  padding: 0.5rem;
  text-transform: uppercase;
  font-weight: bold;
  cursor: ${({ disabled }) => (disabled ? 'auto' : 'pointer')};
  background-color: ${({ disabled }) => (disabled ? '#a9a9a9' : '#468080')};
  border: ${({ disabled }) =>
    disabled ? '1px solid #a9a9a9' : '1px solid #468080'};

  &:focus,
  &:hover:enabled,
  &:active:enabled {
    outline: 2px solid #58bab9;
    background-color: #58bab9;
    outline-offset: -2px;
  }

  ${tablet`
    padding: 1rem;
    margin-top: 1rem;
  `}
`;

const SearchBar = ({
  searchTerm,
  handleSearchTermChange,
  handleSearchSubmit,
  isDisabled,
}: SearchBarProps) => {
  return (
    <SearchBarForm
      onSubmit={(e) => {
        e.preventDefault();
        handleSearchSubmit();
      }}
    >
      <SearchInput
        type="search"
        value={searchTerm}
        onChange={(e) => handleSearchTermChange(e.target.value)}
        placeholder="Search for GitHub repositories"
      />
      <SearchButton type="submit" value="Search" disabled={isDisabled} />
    </SearchBarForm>
  );
};

export default SearchBar;
