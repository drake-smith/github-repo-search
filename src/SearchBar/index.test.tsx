import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import SearchBar from '.';

describe('<SearchBar />', () => {
  const handleSearch = jest.fn();

  test('It renders correctly', () => {
    const tree = renderer
      .create(
        <SearchBar
          searchTerm={''}
          handleSearchSubmit={handleSearch}
          handleSearchTermChange={handleSearch}
          isDisabled={false}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('handleSearch is called correctly', () => {
    render(
      <SearchBar
        searchTerm={''}
        handleSearchSubmit={handleSearch}
        handleSearchTermChange={handleSearch}
        isDisabled={false}
      />
    );
    const buttonElem = screen.getByText('Search');
    fireEvent.click(buttonElem);
    expect(handleSearch).toHaveBeenCalledTimes(1);
  });

  test('Input field value should update correctly', () => {
    render(
      <SearchBar
        searchTerm={''}
        handleSearchSubmit={handleSearch}
        handleSearchTermChange={handleSearch}
        isDisabled={false}
      />
    );
    const searchField = screen.getByLabelText('Search for GitHub repositories');
    fireEvent.change(searchField, { target: { value: 'hello' } });
    expect(handleSearch).toHaveBeenCalled();
  });
});
