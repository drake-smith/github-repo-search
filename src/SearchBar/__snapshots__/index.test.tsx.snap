// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> It renders correctly 1`] = `
<form
  className="sc-bdvvtL jcpPon"
  onSubmit={[Function]}
>
  <input
    className="sc-gsDKAQ kTAeqL"
    onChange={[Function]}
    placeholder="Search for GitHub repositories"
    type="search"
    value="Test"
  />
  <input
    className="sc-dkPtRN cNVLkp"
    disabled={false}
    type="submit"
    value="Search"
  />
</form>
`;
